Userland/Libraries/LibJS/SyntheticModule.cpp

```diff
@@ -0,0 +1,157 @@
+    }
+
+    // 6. Return undefined.
+    // Note: This return value is never visible to the outside so we use void.
+    return {};
+}
+
+// 1.2.3.4 Evaluate ( ), https://tc39.es/proposal-json-modules/#sec-smr-Evaluate
+ThrowCompletionOr<Promise*> JS::SyntheticModule::evaluate(VM& vm)
+{
+    // Note: Has some changes from PR: https://github.com/tc39/proposal-json-modules/pull/13.
+    // 1. Suspend the currently running execution context.
+    // FIXME: We don't have suspend yet.
+
+    // 2. Let moduleContext be a new ECMAScript code execution context.
+    ExecutionContext module_context { vm.heap() };
+
+    // 3. Set the Function of moduleContext to null.
+    // Note: This is the default value.
+
+    // 4. Set the Realm of moduleContext to module.[[Realm]].
+    module_context.realm = &realm();
+
+    // 5. Set the ScriptOrModule of moduleContext to module.
+    module_context.script_or_module = this;
+
+    // 6. Set the VariableEnvironment of moduleContext to module.[[Environment]].
+    module_context.variable_environment = environment();
+
+    // 7. Set the LexicalEnvironment of moduleContext to module.[[Environment]].
+    module_context.lexical_environment = environment();
+
+    // 8. Push moduleContext on to the execution context stack; moduleContext is now the running execution context.
+    vm.push_execution_context(module_context, realm().global_object());
+
+    // 9. Let result be the result of performing module.[[EvaluationSteps]](module).
+    auto result = m_evaluation_steps(*this);
+
+    // 10. Suspend moduleContext and remove it from the execution context stack.
+    vm.pop_execution_context();
+
+    // 11. Resume the context that is now on the top of the execution context stack as the running execution context.
+
+    // 12. Return Completion(result).
+    // Note: Because we expect it to return a promise we convert this here.
+    auto* promise = Promise::create(realm().global_object());
+    if (result.is_error()) {
+        VERIFY(result.throw_completion().value().has_value());
+        promise->reject(*result.throw_completion().value());
+    } else {
+        // Note: This value probably isn't visible to JS code? But undefined is fine anyway.
+        promise->fulfill(js_undefined());
+    }
+    return promise;
+}
+
+// 1.2.2 SetSyntheticModuleExport ( module, exportName, exportValue ), https://tc39.es/proposal-json-modules/#sec-setsyntheticmoduleexport
+ThrowCompletionOr<void> SyntheticModule::set_synthetic_module_export(FlyString const& export_name, Value export_value)
+{
+    // Note: Has some changes from PR: https://github.com/tc39/proposal-json-modules/pull/13.
+    // 1. Return ? module.[[Environment]].SetMutableBinding(name, value, true).
+    return environment()->set_mutable_binding(realm().global_object(), export_name, export_value, true);
+}
+
+NonnullRefPtr<SyntheticModule> SyntheticModule::create_default_export_synthetic_module(Value default_export, Realm& realm, StringView filename)
```

Comment:
```suggestion
// 1.3 CreateDefaultExportSyntheticModule ( defaultExport ), https://tc39.es/proposal-json-modules/#sec-create-default-export-synthetic-module
NonnullRefPtr<SyntheticModule> SyntheticModule::create_default_export_synthetic_module(Value default_export, Realm& realm, StringView filename)
```

Reply:
Or from the PR if you prefer:

```suggestion
// 1.3 CreateDefaultExportSyntheticModule ( defaultExport, realm, hostDefined ), https://tc39.es/proposal-json-modules/#sec-create-default-export-synthetic-module
NonnullRefPtr<SyntheticModule> SyntheticModule::create_default_export_synthetic_module(Value default_export, Realm& realm, StringView filename)
```